collector/src/lib.rs

```diff
@@ -238,21 +170,30 @@ impl<P, Q: QueryType> CollectionJob<P, Q> {
             aggregation_parameter,
         }
     }
+
+    /// Destructure a collection job into its fields.
+    pub fn into_fields(self) -> (CollectionJobId, Query<Q>, P) {
```

Comment:
This method seems like it's roughly equivalent to making the three fields on the struct `pub`. I get that the intent is to enable clients to work out their own serialization scheme, but I don't know if this is much better than three different getter methods. Given that this is public API, I think it might be wiser not to add this and see about adding serialization later, since that wouldn't be a breaking change.

Reply:
I swapped this out for normal getters